pkg/kubectl/cmd/cmd.go

```diff
@@ -258,10 +258,19 @@ __kubectl_custom_func() {
             return
             ;;
         kubectl_logs)
+	    __kubectl_parse_get "deploy" "{{ range .items  }}deploy/{{ .metadata.name }} {{ end }}"
```

Comment:
You'd need to use long names, not aliases. You didn't cover all possible controllers.